### PR TITLE
JAMES-2576 Rework HealthChecks routes status code

### PR DESCRIPTION
**Pull request by chibenwa**
Return 503 upon unhealthy checks.

Return 200 upon degraded checks as service can still be used.

### CODE REVIEW

**Comment by Arsnael** [comment]
```
[c1c48375ae4999f43f63ae822b888839099dc0ce] [ERROR] Failures: 
[c1c48375ae4999f43f63ae822b888839099dc0ce] [ERROR]   GuiceLifecycleHeathCheckTest$Unhealthy.stoppingJamesServerShouldBeUnhealthy:131 1 expectation failed.
[c1c48375ae4999f43f63ae822b888839099dc0ce] Expected status code <500> but was <503>.
```
test this please

**Comment by chibenwa** [comment]
> (and while you are at it you could also add the preview stuff)

Sperated PR.

As part of 3.4.0 or 3.5.0 ?

File: src/site/markdown/server/manage-webadmin.md [hunk 1]
```diff
@@ -99,8 +99,8 @@ Supported health checks include:
 
 Response codes:
 
- - 200: All checks have answered with a Healthy status
- - 500: At least one check have answered with a Unhealthy or Degraded status
+ - 200: All checks have answered with a Healthy or Degraded status. James services can still be used.
+ - 503: At least one check have answered with a Unhealthy or Degraded status
```
Comment (trantienduchn) [inline src/site/markdown/server/manage-webadmin.md]: should `Degraded` be removed?

**Comment by chibenwa** [comment]
`MemoryVacationIntegrationTest>VacationIntegrationTest.jmapVacationShouldNotSendNotificationTwice:241 » ConditionTimeout`

**Comment by rouazana** [comment]
> As part of 3.4.0 or 3.5.0 ?

3.5.0, 3.4.0 is already out ^^

**Comment by chibenwa** [comment]
```
[a1f1e85f230ed28471dc02e35f35833add93466d] [ERROR] Failures: 
[a1f1e85f230ed28471dc02e35f35833add93466d] [ERROR]   EventDeadLettersIntegrationTest.failedEventShouldBeStoredInDeadLetterUnderItsGroupId:212 1 expectation failed.
[a1f1e85f230ed28471dc02e35f35833add93466d] JSON path . doesn't match.
[a1f1e85f230ed28471dc02e35f35833add93466d] Expected: a collection with size <1>
[a1f1e85f230ed28471dc02e35f35833add93466d]   Actual: []
```

test this please

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**